File: src/BuildingBlocks/nodejs/framework/bus/abstratctions/iBus.ts
import { IntegrationEvent, Message,IStartWithMessage, Command } from '../../infrastructure';
import { IIntegrationEventHandler } from "./iIntegrationEventHandler";
import { Saga} from "./saga";

export interface ICommandBus {
     Send<T extends Command>(command: T): void;    
     RegisterSaga<T extends Saga>(startWithMessage:string,saga:T): void;
     RegisterHandler<T>():void;
}

export interface IEventBus {
    Publish(event: IntegrationEvent): void;
    Subscribe<T extends IntegrationEvent, TH extends IIntegrationEventHandler<T>>(): void;
    Unsubscribe<T extends IntegrationEvent, TH extends IIntegrationEventHandler<T>>(): void;
}

File: src/BuildingBlocks/dotnet/Framework/Bus/Abstractions/IBus.cs
using Microsoft.EntityFrameworkCore;
using Monad.EDukaan.Framework.Common.Entities;
using Monad.EDukaan.Framework.Persistence.Domain.Interfaces;
using System.Collections.Generic;
using System.Linq;

namespace Monad.EDukaan.Framework.Bus.Abstractitons
{
    public interface ICommandBus
    {
        void Send<T>(T command) where T : Command;
        void RegisterSaga<T>(T saga) where T : Saga;
        void RegisterHandler<T>();
    }

    public interface IEventBus
    {
        void Publish<T>(T command) where T : IntegrationEvent;
        void Subscribe<T, TH>() where T : IntegrationEvent, TH, IIntegrationEventHandler<T>;
        void UnSubscribe<T, TH>() where T : IntegrationEvent, TH, IIntegrationEventHandler<T>;
    }
}